sdk/core/core-amqp/src/auth/sas.ts

```diff
@@ -78,7 +84,98 @@ export class SharedKeyCredential {
    * @param {string} connectionString - The EventHub/ServiceBus connection string
    */
   static fromConnectionString(connectionString: string): SharedKeyCredential {
-    const parsed = parseConnectionString<ServiceBusConnectionStringModel>(connectionString);
-    return new SharedKeyCredential(parsed.SharedAccessKeyName, parsed.SharedAccessKey);
+    const parsed = parseConnectionString<
+      ServiceBusConnectionStringModel & { SharedAccessSignature: string }
+    >(connectionString);
+
+    if (parsed.SharedAccessSignature == null) {
+      return new SharedKeyCredential(parsed.SharedAccessKeyName, parsed.SharedAccessKey);
+    } else {
+      return new SharedAccessSignatureCredential(parsed.SharedAccessSignature);
+    }
+  }
+}
+
+/**
+ * A credential that takes a SharedAccessSignature of two forms:
+ * `SharedAccessSignature sr=<resource>&sig=<signature>&se=<expiry>&skn=<keyname>`
+ * or
+ * `sr=<resource>&sig=<signature>&se=<expiry>&skn=<keyname>`
+ *
+ * @internal
+ * @ignore
+ */
+export class SharedAccessSignatureCredential extends SharedKeyCredential {
+  private _accessToken: AccessToken;
+
+  /**
+   * @param sharedAccessSignature A shared access signature of the form `sr=<resource>&sig=<signature>&se=<expiry>&skn=<keyname>`
+   */
+  constructor(sharedAccessSignature: string) {
+    super("", "");
+
+    const {
+      keyName,
+      expiry: expiresOnTimestamp,
+      token
+    } = SharedAccessSignatureCredential.extractSasProperties(sharedAccessSignature);
+
+    this.keyName = keyName;
+    this.key = "<none>";
+    this._accessToken = {
+      token,
+      expiresOnTimestamp
+    };
+
+    this.renewable = false;
```

Comment:
Why is this always false?

Reply:
We can't ever renew one of these - it's just a SAS signature with a predefined expiry date and AFAIK you can't generate a new one with just this signature (you need the key).